### PR TITLE
[DEV-10893] - create request interactive mode

### PR DESCRIPTION
**Pull request by tomerstein**

### CODE REVIEW

File: pkg/inputs/list_select/model.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package listselect
+
+import "github.com/charmbracelet/bubbles/list"
+
+type SelectInput[T any] struct {
```
Comment (naorgu) [inline pkg/inputs/list_select/model.go:5]: the generic implementation is nice, but I wonder if it's not too generic in a way that it causes too much overhead and complexity. let's discuss

File: pkg/inputs/text_input/keys.go
Comment (naorgu) [inline pkg/inputs/text_input/keys.go:1]: this package also should be inside `interactive`

File: pkg/styles/titles.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package styles
```
Comment (naorgu) [inline pkg/styles/titles.go:1]: this should also be inside `interactive` right ?

**Review by naorgu — COMMENTED** [review]
mega cool shmul

**Comment by naorgu** [comment]
see this
```
dist/apono_darwin_arm64/apono requests create -i sdaf
Error: if any flags in the group [integration resource-type resources permissions] are set they must all be set; missing [permissions resource-type resources]
```

File: pkg/styles/titles.go (hunk 2 above)
Comment (tomerstein) [reply pkg/styles/titles.go:1]: no this package only change text style, can be used in regular commands

File: pkg/interactive/justification.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package interactive
+
+import (
+	textinput2 "github.com/apono-io/apono-cli/pkg/interactive/inputs/text_input"
```
Comment (naorgu) [inline pkg/interactive/justification.go:4]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/text_input"
```

File: pkg/interactive/permissions.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package interactive
+
+import (
+	"context"
+	"fmt"
+
+	listselect "github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```
Comment (naorgu) [inline pkg/interactive/permissions.go:7]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```

File: pkg/interactive/resource_types.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package interactive
+
+import (
+	"context"
+	"fmt"
+
+	listselect "github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```
Comment (naorgu) [inline pkg/interactive/resource_types.go:7]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```

File: pkg/interactive/resources.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
+package interactive
+
+import (
+	"context"
+	"fmt"
+
+	listselect "github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```
Comment (naorgu) [inline pkg/interactive/resources.go:7]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```

File: pkg/interactive/bundles.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
+package interactive
+
+import (
+	"context"
+	"fmt"
+
+	listselect "github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```
Comment (naorgu) [inline pkg/interactive/bundles.go:7]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```

File: pkg/interactive/integrations.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package interactive
+
+import (
+	"context"
+	"fmt"
+
+	listselect "github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```
Comment (naorgu) [inline pkg/interactive/integrations.go:7]:
```suggestion
	"github.com/apono-io/apono-cli/pkg/interactive/inputs/list_select"
```

File: pkg/interactive/inputs/list_select/model.go [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package listselect
+
+import "github.com/charmbracelet/bubbles/list"
+
+type SelectOption struct {
+	ID     string
+	Label  string
+	Filter string
```
Comment (naorgu) [inline pkg/interactive/inputs/list_select/model.go]: as we talked probably we dont need the Filter

File: pkg/styles/titles.go [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+	for i, name := range names {
+		if i != len(names)-1 {
+			result += selectedItemsColor.Sprint(name) + ", "
+		} else {
+			result += selectedItemsColor.Sprint(name)
+		}
+	}
+
+	return result
+}
```
Comment (naorgu) [inline pkg/styles/titles.go:38]: maybe simpler

```suggestion
func joinNames(names []string) string {
	var coloredNames []string
	for _, name := range names {
		coloredNames = append(coloredNames, selectedItemsColor.Sprint(name))
	}
	
	return strings.Join(coloredNames, ", ")
}
```

**Review by naorgu — APPROVED** [review]
fabulous

**Pull request merged**